core/src/main/java/org/aerogear/mobile/core/metrics/Metrics.java

```diff
@@ -3,24 +3,26 @@
 import org.json.JSONObject;
 
 /**
- * Interface for a record to be published to the metrics service
- * 
+ * Record to be published to the metrics service
+ *
+ * WARN: Not intended by direct usage outside of the SDK
+ *
  * @param <T> A type that can be serialized into a {@link JSONObject} value
  */
-public interface Metrics<T> {
+public abstract class Metrics<T> {
```

Comment:
So I'm -1 on this change.  I think it might be better (and less hacky) if our publishers are declared final and we do instance of enforcement of metrics implementations at that level.

Reply:
Okay, I'll revert this back to the regular interface.